cpp/src/arrow/python/arrow_to_pandas.cc

```diff
@@ -280,6 +280,9 @@ class PandasBlock {
 
 template <typename T>
 inline const T* GetPrimitiveValues(const Array& arr) {
+  if (arr.length() == 0) {
```

Comment:
Unrelated to this PR, but is there a policy on documenting/commenting internal functions?

Reply:
There's no policy per se, but since there are more of us working on this codebase now, we should be more diligent about explaining what internal APIs are doing in comments